### DISCUSSION TITLE
Random Instant Serverless Function Timeouts (on all API routes)

### DISCUSSION THREAD

**Discussion by AAlber** [vercel/community#1442] [Help] [answered]
Hi,

We've been dealing with this issue in our t3 application for over a week now. Here is the error we are getting in the Vercel function logs:  

`START RequestId: 3e875969-afce-4409-a7ae-96efc99ff603 Version: $LATEST 2023-01-31T13:43:45.065Z d22a6e7d-0ef6-4aa2-828e-ab70ceb99f02 ERROR Unhandled Promise Rejection  {\"errorType\":\"Runtime.UnhandledPromiseRejection\",\"errorMessage\":\"Error: timeout of 3000ms exceeded\",\"trace\":[\"Runtime.UnhandledPromiseRejection: Error: timeout of 3000ms exceeded\",\" at process.<anonymous> (file:///var/runtime/index.mjs:1194:17)\",\" at process.emit (node:events:525:35)\",\" at process.emit (node:domain:489:12)\",\" at emit (node:internal/process/promises:149:20)\",\" at processPromiseRejections (node:internal/process/promises:283:27)\",\" at process.processTicksAndRejections (node:internal/process/task_queues:96:32)\"]} [ERROR] [1675172625066] LAMBDA_RUNTIME Failed to post handler success response. Http response code: 400. RequestId: 3e875969-afce-4409-a7ae-96efc99ff603 Error: Runtime exited without providing a reason Runtime.ExitError END RequestId: 3e875969-afce-4409-a7ae-96efc99ff603 REPORT RequestId: 3e875969-afce-4409-a7ae-96efc99ff603. Duration: 12.50 ms Billed Duration: 13 ms Memory Size: 1024 MB Max Memory Used: 143 MB"` 

This only happens to Vercel deployed branches.

Notice how it says that it is a 3000ms timeout error but the duration is only 12.50ms. 

More details:
- All necessary variables (like required user id's for API calls) are provided
- The failing API calls succeed most of the time but about 1/5 times they fail. 
- Includes all our API calls 

Example failing API call:
```
import type { NextApiRequest, NextApiResponse } from "next";
import { prisma } from "../../../server/db/client";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  if (req.method === "POST") {
    try {
      const data = JSON.parse(req.body);
      const update = await prisma.user.update({
        where: { id: data.id },
        data: data.data,
      });
      res.json(update);
    } catch (e: any) {
      res.status(500).json({ message: e.message });
      throw new Error("Could't update user:"+ e.message)
    }
  }
}
```
**Note**: The error messages in the catch block are printed nowhere. 

This is more or less how we call all our serverless functions:
```
async function updateUserCurrentInstitution({ userId, institutionId }) {
  const response = await fetch(api.updateUser, {
    method: "POST",
    body: JSON.stringify({
      id: userId,
      data: {
        currentInstitution: institutionId,
      },
    }),
  });
  if (!response.ok) {
    throw new Error(response.statusText);
  }
  return response;
}
```

Any help would be greatly appreciated !

**Comment by xjustleoo** [comment]
same issue for me. Includes most of my API calls!

**Comment by keithlimmontajes** [comment]
+1

**Comment by amyegan** [comment, marked as answer]
_Edit: It was caused by a third party library - https://github.com/vercel/vercel/discussions/9350#discussioncomment-4839480_

---
Thanks for reporting these issues everyone. Do any of you have example deployment links that I can use to try to see the problem? Since it's intermittent, I think my best chance to see it happen is with an app that has already run into the error.

@AAlber It's interesting that the catch block message isn't logged anywhere. If you add some `console.log` statements at the beginning of your try and catch blocks, do you see both of them show up in your function logs?

**Reply by AAlber** [reply]
I actually finally found out that this issue was caused by a third party library which kept running into rate limits. Sorry to have bothered you for no reason !

**Reply by amyegan** [reply]
No problem! I'm glad you found a solution 😄
